TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs

```diff
@@ -9,11 +9,29 @@ public UserMenuHeaderButtonPageComponent(IWebDriver driver)
             this.driver = driver;
         }
 
-        private IWebElement _userMenu => driver.FindElement(By.XPath("//button/span[@class='MuiIconButton-label-2477']/div"));
+        private By _userMenu = By.XPath("//div[contains(@class, 'UserMenu-avatar')]");
 
-        public UserMenuDropDownListPageComponent ClickUserMenuButton()
+        public UserMenuDropDownListPageComponent ClickUserMenuButton(int timeToWait)
         {
-            _userMenu.Click();
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(timeToWait));
```

Comment:
I think you can use method of base component class

Reply:
agree, I've just not changed this code here